File: chillflix-patches/mainsite/verify_hero_badges.py
#!/usr/bin/env python3
import re
import urllib.request

req = urllib.request.Request(
    "http://127.0.0.1:3000/",
    headers={"Host": "www.chillflix.lol", "User-Agent": "Mozilla/5.0"},
)
with urllib.request.urlopen(req, timeout=30) as res:
    h = res.read().decode("utf-8", "ignore")

print("len", len(h))
for n in [
    "home-hero-badges",
    "home-hero-badge-today",
    "home-hero-badge-hot",
    "home-hero-badge-new",
    "Newly launched",
    "HOT",
    "Today",
]:
    print(("OK" if n in h else "MISS"), n)

m = re.search(r'src="(https://[^"]+)"[^>]*(?:logo|alt=)|(?:logo)[^>]*src="(https://[^"]+)"', h, re.I)
# Prefer wsrv/tmdb logo paths near home-hero
logos = re.findall(r'https://[^"\']+(?:/t/p/|wsrv\.nl)[^"\']+', h)
print("logo_candidates", len(logos))
if logos:
    print("logo0", logos[0][:160])

m2 = re.search(r"home-hero-badge-rank[^>]*>\s*#?(\d+)", h)
print("rank", m2.group(1) if m2 else None)

File: chillflix-patches/newsite/verify_hero_badges.php
<?php
$_SERVER['REQUEST_METHOD'] = 'GET';
$_SERVER['HTTP_HOST'] = 'www.chillflix.lol';
$_SERVER['REQUEST_URI'] = '/newsite/home';
$_SERVER['SCRIPT_NAME'] = '/newsite/index.php';
chdir('/var/www/chillflix-newsite/public');
ob_start();
include 'index.php';
$h = ob_get_clean();
echo 'LEN=' . strlen($h) . PHP_EOL;
foreach ([
    'hero-logo',
    'hero-badges',
    'hero-badge-today',
    'hero-badge-hot',
    'hero-badge-new',
    'Newly launched',
    'HOT',
    'Today',
] as $n) {
    echo (str_contains($h, $n) ? 'OK' : 'MISS') . " $n" . PHP_EOL;
}
if (preg_match('/hero-logo[\s\S]{0,200}<img[^>]+src="([^"]+)"/', $h, $m)) {
    echo 'LOGO_SRC=' . $m[1] . PHP_EOL;
}
if (preg_match('/hero-badge-rank">([^<]+)</', $h, $m)) {
    echo 'RANK=' . $m[1] . PHP_EOL;
}
